core/tests/test_utils.py

```diff
@@ -2280,37 +2281,66 @@ def assert_same_list_elements(self, phrases, stdout):
 GenericTestBase = AppEngineTestBase
 
 
-class GenericEmailTestBase(GenericTestBase):
-    """Base class for tests requiring email services."""
+class EmailMessageMock(python_utils.OBJECT):
+    """Mock for core.platform.models email services messages."""
+
+    def __init__(
+            self, sender_email, recipient_email, subject, plaintext_body,
+            html_body, bcc=None, reply_to=None, recipient_variables=None):
+        """Inits a mock email message with all the necessary data.
+
+        Args:
+            sender_email: str. The email address of the sender. This should be
+                in the form 'SENDER_NAME <SENDER_EMAIL_ADDRESS>' or
+                'SENDER_EMAIL_ADDRESS'. Must be utf-8.
+            recipient_email: str. The email address of the recipient.
+                Must be utf-8.
+            subject: str. The subject line of the email, Must be utf-8.
+            plaintext_body: str. The plaintext body of the email. Must be utf-8
+            html_body: str. The HTML body of the email. Must fit in a datastore
+                entity. Must be utf-8.
+            bcc: list(str)|str|None. Optional argument. List of bcc emails,
```

Comment:
Do not allow polymorphic types please. Either list(str) or str, but not both.

Ditto elsewhere.

Reply:
Done